frontend/src/pages/GeneralSettings/Chats/index.jsx

```diff
@@ -7,29 +7,60 @@ import useQuery from "@/hooks/useQuery";
 import ChatRow from "./ChatRow";
 import showToast from "@/utils/toast";
 import System from "@/models/system";
-
-const PAGE_SIZE = 20;
+import { CaretDown } from "@phosphor-icons/react";
 export default function WorkspaceChats() {
+  const [showMenu, setShowMenu] = useState(false);
+  const [exportType, setExportType] = useState("csv");
+  const menuRef = useRef();
+  const openMenuButton = useRef();
+
+  const exportOptions = {
+    csv: { mimeType: "text/csv", fileExtension: "csv" },
+    json: { mimeType: "application/json", fileExtension: "json" },
+    jsonl: { mimeType: "application/jsonl", fileExtension: "jsonl" },
+  };
   const handleDumpChats = async () => {
-    const chats = await System.exportChats();
+    const chats = await System.exportChats(exportType);
     if (chats) {
-      const blob = new Blob([chats], { type: "application/jsonl" });
+      const { mimeType, fileExtension } = exportOptions[exportType];
+      const blob = new Blob([chats], { type: mimeType });
       const link = document.createElement("a");
       link.href = window.URL.createObjectURL(blob);
-      link.download = "chats.jsonl";
+      link.download = `chats.${fileExtension}`;
       document.body.appendChild(link);
       link.click();
       window.URL.revokeObjectURL(link.href);
       document.body.removeChild(link);
       showToast(
-        "Chats exported successfully. Note: Must have at least 10 chats to be valid for OpenAI fine tuning.",
+        `Chats exported successfully as ${fileExtension.toUpperCase()}. Note: Must have at least 10 chats to be valid for OpenAI fine tuning.`,
         "success"
       );
     } else {
       showToast("Failed to export chats.", "error");
     }
   };
 
+  const toggleMenu = () => {
+    setShowMenu(!showMenu);
+  };
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (
+        menuRef.current &&
+        !menuRef.current.contains(event.target) &&
+        !openMenuButton.current.contains(event.target)
+      ) {
+        setShowMenu(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
```

Comment:
Does this give you an error in console? It does for me sometimes

Reply:
Hmm I don't get any errors in console with this for any case.